pgnReader.py

```diff
@@ -0,0 +1,38 @@
+'''
+    path - path to .pgn file
+    maxGame - maximum number of games to parse
+'''
+def pgnParse(path,maxGame=None):
+    result=[{}]
+    with open(path,'r') as f:
+        for line in f:
+            #tags begin with "["
+            #Movetext begin with "1."
+            if(line[0]=="["):
+                j=line.split(" ")
+                result[-1][j[0][1:]]=j[1][1:-2]
+            elif(line[0]=="1"):
+                game=[]
+                for j in line.split(". ")[1:]:
+                    #there are several kinds of Movetext notation, so also several kinds of split
+                    if(" {" in j):
+                        moves=j.split(" {")
+                        for i in moves[:-1]:
+                            game.append(i.split(" ")[-1].replace("!","").replace("!?","").replace("?",""))
+                    else:
+                        moves=j.split(" ")
+                        game.append(moves[0])
+                        if(("\n" in moves[1])==False and (moves[1].isdigit())==False):
+                            game.append(moves[1].replace("!","").replace("!?","").replace("?",""))

```

Comment:
maybe factor removing '!', '!?', '?' to a different function? this way you can *name the operation* using a descriptive function name and therefore  make the code clear without comment